### ISSUE TITLE
Pure Java implementation

### ISSUE THREAD

**Issue by koppor** [michel-kraemer/citeproc-java#92]
The current version is implemented by using GraalVM and an existing JavaScript implementation. This causes troubles with other JVMs, such as the [Eclipse OpenJ9 VM](https://www.eclipse.org/openj9/): https://github.com/eclipse/openj9/issues/11555.

This issue tries to make the desire of a pure Java implementation explicit. Maybe, it helps to shape a project tackling this wish.

Side note: There are transpilers from Java to JavaScript available (e.g., [jsweet](http://www.jsweet.org/) or [j2cl](https://www.infoq.com/news/2019/05/j2cl-java-javascript-transpiler/)), however, there is no JavaScript to Java transpiler (cf. https://awesomeopensource.com/projects/transpiler).

Also refs https://github.com/JabRef/jabref/issues/5622#issuecomment-590334280.

**Comment by michel-kraemer** [comment]
@koppor The pure Java version already exists. It's called "experimental" mode and is available in the latest snapshot. As far as I know we've already integrated this snapshot in JabRef and enabled experimental mode, haven't we? If yes, and if the results look OK, I can of course release a final version of it if you wish.

**Comment by koppor** [comment]
I totally forgot about the status. Yes, we integrated it and had no complains.

I digged into the dependencies. The tree shows as follows. `org.graalvm` should be obsolete now, shouldn't it? Does it require a separte branch to remove that (e.g., https://github.com/michel-kraemer/citeproc-java/blob/5d119ecde0e21b53e217be33730fa3dd8901c99c/citeproc-java/build.gradle#L25) or should we do it with gradle dependency management magic? I would prefer to remove the whole depedency in the current implementation to have a clean state. Maybe, a call for releasing a `3.0` version, even though it's "only" an internal dependency change?

```
+--- de.undercouch:citeproc-java:2.1.0-SNAPSHOT
|    +--- org.apache.commons:commons-lang3:3.9
|    +--- org.apache.commons:commons-text:1.8
|    |    \--- org.apache.commons:commons-lang3:3.9
|    +--- org.jbibtex:jbibtex:1.0.17
|    +--- org.antlr:antlr4-runtime:4.7.2 -> 4.9
|    +--- org.graalvm.sdk:graal-sdk:19.2.1
|    \--- org.graalvm.js:js:19.2.1
|         +--- org.graalvm.regex:regex:19.2.1
|         |    \--- org.graalvm.truffle:truffle-api:19.2.1
|         |         \--- org.graalvm.sdk:graal-sdk:19.2.1
|         +--- org.graalvm.truffle:truffle-api:19.2.1 (*)
|         +--- org.graalvm.sdk:graal-sdk:19.2.1
|         +--- org.ow2.asm:asm:6.2.1
|         +--- org.ow2.asm:asm-tree:6.2.1
|         |    \--- org.ow2.asm:asm:6.2.1
|         +--- org.ow2.asm:asm-analysis:6.2.1
|         |    \--- org.ow2.asm:asm-tree:6.2.1 (*)
|         +--- org.ow2.asm:asm-commons:6.2.1
|         |    +--- org.ow2.asm:asm:6.2.1
|         |    +--- org.ow2.asm:asm-tree:6.2.1 (*)
|         |    \--- org.ow2.asm:asm-analysis:6.2.1 (*)
|         +--- org.ow2.asm:asm-util:6.2.1
|         |    +--- org.ow2.asm:asm:6.2.1
|         |    +--- org.ow2.asm:asm-tree:6.2.1 (*)
|         |    \--- org.ow2.asm:asm-analysis:6.2.1 (*)
|         \--- com.ibm.icu:icu4j:62.1
```

**Comment by michel-kraemer** [comment]
My proposal would be to release both 2.1.0 (still with GraalVM for people who still want to use it) and 3.0 (with the pure Java experimental mode promoted to the default and GraalVM removed). If you like I can try to release the new version as soon as possible (maybe on the weekend or some time next week).

**Comment by koppor** [comment]
I would be very happy if you could release a `3.0.0-SNAPSHOT`. This would really simplify our solution. And probably also unblock https://github.com/JabRef/jabref/pull/7126#issuecomment-735433801 (caused by too many dependencies in our module)

**Comment by michel-kraemer** [comment]
I just published a first `3.0.0-SNAPSHOT` without GraalVM dependencies:
https://oss.sonatype.org/content/groups/public/de/undercouch/citeproc-java/3.0.0-SNAPSHOT/

Please test and let me know if there are any problems.

**Comment by koppor** [comment]
It's in JabRefs main branch now. In case we experience issues, we will report back. Thank you for the quick action taken! 🎉

**Comment by koppor** [comment]
@michel-kraemer Would it be possible to create a 3.0.0-alpha release or similar? Since the constructor changed in the SNAPSHOT release (which is OK for a snapshot), there were build issues at JabRef. See the discussions at https://github.com/JabRef/jabref/pull/7514. I think, it might be too early for a 3.0.0 release, but maybe a tagged release could be possible? This would help us to get a reproducible build.

**Comment by michel-kraemer** [comment]
@koppor Sure, no problem. I've just released version 3.0.0-alpha.1
